app/actions/views/channel.js

```diff
@@ -344,21 +342,32 @@ export function selectDefaultChannel(teamId) {
         if (channelId) {
             dispatch(setChannelDisplayName(''));
             dispatch(handleSelectChannel(channelId));
-            dispatch(markChannelAsRead(channelId));
         }
     };
 }
 
-export function handleSelectChannel(channelId) {
+export function handleSelectChannel(channelId, skipChannelViewedAt = false) {
     return async (dispatch, getState) => {
-        const {currentTeamId} = getState().entities.teams;
-
-        dispatch(setLoadMorePostsVisible(true));
+        const state = getState();
+        const currentTeamId = getCurrentTeamId(state);
+        const currentChannelId = getCurrentChannelId(state);
+        const sameChannel = channelId === currentChannelId;
+        let member = getMyChannelMember(state, channelId);
 
-        loadPostsIfNecessaryWithRetry(channelId)(dispatch, getState);
-        selectChannel(channelId)(dispatch, getState);
+        if (skipChannelViewedAt) {
+            const lastPost = getLastPostPerChannel(state, channelId);
+            member = {
+                last_viewed_at: lastPost[channelId].create_at,
+            };
+        }
 
+        dispatch(setLoadMorePostsVisible(true));
+        dispatch(loadPostsIfNecessaryWithRetry(channelId));
         dispatch(batchActions([
+            {
+                type: ChannelTypes.SELECT_CHANNEL,
```

Comment:
before we were using `dispatch(selectChannel(channelId))` but that is a thunk and I wanted to batch all this.

Reply:
perhaps best if we change the redux action from being a thunk to be pure

Reply:
I think you might be able to batch thunk actions if we have the middleware set up in the right order. It looks like it would actually be trivial to make `selectChannel` a non-thunk action though